### PR TITLE
feat: Reports warnings on failed kube deploy, fixes error out

### LINKED ISSUE #2621: Layout broken for a "Play pod or Container from a Kubernetes YAML" after sucessfull pod creation
### Bug description

I got broken layout, almost endless json response editor under "Done" button after "Play Kubernetes YAML".
broken-playout-play-kube-yaml

### Operating system

Fedora 37

### Version

1.0.x

### Steps to reproduce

1. Pull httpd-24 (quay.io/centos7/httpd-24-centos7) image
2. Start container
3. Create a kube from httpd container
4. Play Kubernetes YAML with create yaml, choose podman runtime

My configuration is that I also have openshift local, kind and sandbox resources started. kube context is `crc-admin` right now. 

### Relevant log output

```shell
I am not sure if those console log are relevant:

97 index.ts:1327 Uncaught Error: Cannot read properties of undefined (reading 'log')
    at EventEmitter.<anonymous> (<anonymous>:2:7777)
    at EventEmitter.emit (VM159 sandbox_bundle:2:35718)
    at Object.onMessage (VM159 sandbox_bundle:2:47921)
(anonymous) @ index.ts:1327
emit @ VM159 sandbox_bundle:2
onMessage @ VM159 sandbox_bundle:2
```

### Additional context

_No response_

### LINKED ISSUE #3033: Play Kubernetes YAML does not error out despite container errors on creation.
### Bug description

Screenshot 2023-06-27 at 2 31 56 PM

See "ContainerErrors".

We are able to successfully deploy, but when the containers are created, we encounter an error (such as a port being taken) but we do not error our or warn the user.

Because this error occurs *after* creation (we create and THEN start), we should at least warn the user that the container has been created, but the port is unavailable.

### Operating system

mac M1

### Installation Method

None

### Version

1.1.0

### Steps to reproduce

Using the following YAML:

```
# Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-4.5.1
apiVersion: v1
kind: Pod
metadata:
  annotations:
    io.podman.annotations.ulimit: nofile=524288:524288,nproc=7248:7248
  creationTimestamp: "2023-06-27T17:47:37Z"
  labels:
    app: reverentrobinson-pod
  name: reverentrobinson-pod
spec:
  containers:
  - image: docker.io/cdrage/helloworld:latest
    name: reverentrobinson
    ports:
    - containerPort: 80
      hostPort: 9000
    tty: true


```

1. Do "Deploy Kubernetes YAML"
2. Select Podman container engine
3. See the json output where there is errors, but there is no visible errors within podman desktop UI

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
feat: Reports warnings on failed kube deploy, fixes error out

### What does this PR do?

This updates the Play Kubernetes YAML section similar to the output of "Deploy to Kubernetes".

* Reports warnings when encountering container errors when using Play
  Kubernetes
* Fixes the error / warning output that was not appearing correctly.

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/6422176/3d7a2a1a-68c6-4623-9470-ec2b74cad682

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/2621
Closes https://github.com/containers/podman-desktop/issues/3033

### How to test this PR?

1. Press Play Kubernetes
2. Use the following YAML:

```
# Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-4.5.1
apiVersion: v1
kind: Pod
metadata:
  annotations:
    io.podman.annotations.ulimit: nofile=524288:524288,nproc=7248:7248
  creationTimestamp: "2023-06-27T17:47:37Z"
  labels:
    app: reverentrobinson-pod
  name: reverentrobinson-pod
spec:
  containers:
  - image: docker.io/cdrage/helloworld:latest
    name: reverentrobinson
    ports:
    - containerPort: 80
      hostPort: 9000
    tty: true
```
3. Should appear correctly.
4. Try deploying again / you'll now get the correct erroring / warning.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
TODO: Need to add tests (trying to figure out how to mock a `.yaml` being passed in as a selected file)

**Comment by benoitf** [comment]
@cdrage I think you can just mock `window.openFileDialog` and then you mock `window.playKube` as well

**Review by jeffmaury — DISMISSED** [review]
I thought lint was checked at commit time

**Comment by benoitf** [comment]
@jeffmaury it is unless you bypass git hooks :-) (`-n, --[no-]verify` option on git)

**Comment by cdrage** [comment]
> @jeffmaury it is unless you bypass git hooks :-) (`-n, --[no-]verify` option on git)

Haha yup, I had to bypass the hook so I can push out the code for those to test it out :) Lint is fixed now.

**Comment by cdrage** [comment]
Tests added 💯 ready for another review.

**Review by lstocchi — COMMENTED** [review]
By running your example i get the old error 
`Error: (HTTP code 500) server error - playing YAML file: adding pod to state: name "reverentrobinson-pod" is in use: pod already exists` . 

By debugging it, it gets in the catch and it skips your code. Do you have another yaml file to test it?

**Comment by cdrage** [comment]
> By running your example i get the old error `Error: (HTTP code 500) server error - playing YAML file: adding pod to state: name "reverentrobinson-pod" is in use: pod already exists` .
> 
> By debugging it, it gets in the catch and it skips your code. Do you have another yaml file to test it?

That's intentional? I believe you still have the old pod running when trying to deploy.

**Comment by lstocchi** [comment]
> That's intentional? I believe you still have the old pod running when trying to deploy.

I thought I followed your instruction. Have i misunderstood them? 🤔 

Click play, select the yaml, deploy, everything works fine, repeat to see the error. No?

**Comment by cdrage** [comment]
> > That's intentional? I believe you still have the old pod running when trying to deploy.
> 
> I thought I followed your instruction. Have i misunderstood them? 🤔
> 
> Click play, select the yaml, deploy, everything works fine, repeat to see the error. No?

Yup! It's intentional. The error is appearing. 

https://github.com/containers/podman-desktop/issues/3033 = It's showing the error.

https://github.com/containers/podman-desktop/issues/2621 = no more layout being broken.

Playing it twice induces the error and shows what this PR is fixing.

File: packages/renderer/src/lib/kube/KubePlayYAML.svelte [hunk 1]
```diff
@@ -57,8 +60,24 @@ async function playKubeFile(): Promise<void> {
...
+
         // remove the null values from the result
         playKubeResultRaw = JSON.stringify(removeEmptyOrNull(result), null, 2);
+        playKubeResultJSON = JSON.parse(playKubeResultRaw);
+
+        // If there are container errors, that means that it was *able* to create the container
+        // but if failed to start. We will add this to the "warning" section as we were able to create the
+        // We add this with comma deliminated errors
+        if (playKubeResultJSON.Pods.length > 0) {
+          const containerErrors = playKubeResultJSON.Pods.filter(pod => pod.ContainerErrors.length > 0);
```
Comment (lstocchi) [inline packages/renderer/src/lib/kube/KubePlayYAML.svelte]: I got an error here. pod.ContainerErrors was undefined
```
error playing kube file TypeError: Cannot read properties of undefined (reading 'length')
    at KubePlayYAML.svelte:72:93
    at Array.filter (<anonymous>)
    at playKubeFile (KubePlayYAML.svelte:72:59)
```

**Review by lstocchi — COMMENTED** [review]
Ok I think i got it working by editing your yaml. Works fine, also the issue i saw with the broken layout is fixed.
Just a problem with a possible undefined value. For the rest looks good

File: packages/renderer/src/lib/kube/KubePlayYAML.svelte (hunk 1 above)
Comment (cdrage) [reply packages/renderer/src/lib/kube/KubePlayYAML.svelte]: Ah thank you! I'll fix this. We should check to see that containerErrors actually exists before filtering.

**Comment by cdrage** [comment]
@lstocchi Thanks @lstocchi ! can you check the newest commit? I've added a change to that single line of code.

**Review by lstocchi — APPROVED** [review]
LGTM, works fine! Thanks @cdrage 🚀

**Pull request merged**